src/git/zcl_abapgit_git_transport.clas.abap

```diff
@@ -7,17 +7,19 @@ CLASS zcl_abapgit_git_transport DEFINITION
 
 * remote to local
     CLASS-METHODS upload_pack
-      IMPORTING
-        !iv_url         TYPE string
-        !iv_branch_name TYPE string
-        !iv_deepen      TYPE abap_bool DEFAULT abap_true
-        !it_branches    TYPE zif_abapgit_definitions=>ty_git_branch_list_tt OPTIONAL
-      EXPORTING
-        !et_objects     TYPE zif_abapgit_definitions=>ty_objects_tt
-        !ev_branch      TYPE zif_abapgit_definitions=>ty_sha1
-        !eo_branch_list TYPE REF TO zcl_abapgit_git_branch_list
-      RAISING
-        zcx_abapgit_exception .
+      IMPORTING io_repo         TYPE REF TO zcl_abapgit_repo OPTIONAL
+                iv_url          TYPE string
+                iv_branch_name  TYPE string
+                iv_commit_hash  TYPE zif_abapgit_definitions=>ty_sha1 OPTIONAL
+                iv_deepen       TYPE abap_bool DEFAULT abap_true
+                iv_deepen_level TYPE numc2 DEFAULT '1'
```

Comment:
both parameters iv_deepen and iv_deepen_level are not needed, suggest if deepen = 0 then deepen is not enabled